### ISSUE TITLE
How to set metric for lightgbm?

### ISSUE THREAD

**Issue by magicdict** [dotnet/machinelearning#3884]
I compare the python lightgbm and ml.net,some parm can't set.
```python
params = {
    "boosting_type": "gbdt",
    "objective": "binary",
    "metric": "binary_logloss",
    "num_leaves": 32,
    "learning_rate": 0.05,
    "feature_fraction": 0.9,
    "bagging_fraction": 0.8,
    "bagging_freq": 5,
    "verbose": 1,
}
```
metric is one of miss parm.

By the way,how to display the progress of training lightgbm at console like python?

**Comment by magicdict** [comment]
I found LightGbmRegressionTrainer.Options to set metric,but how to set valid_sets for lightgbm?

```python
gbm = lgb.train(
    params,
    lgb_train,
    num_boost_round=5000,
    valid_sets=lgb_eval,
    early_stopping_rounds=50,
    verbose_eval=50,
)
```

**Comment by rauhs** [comment]
You can pass it to the `Fit(train, validationDv)` call.

**Comment by magicdict** [comment]
> You can pass it to the `Fit(train, validationDv)` call.

Thankyou for your reply.
When only use lightgbm trainer,I can find Fit(train,validationData).
But I only find Fit(train) method when trainer at the end of pipeline. 

```csharp
var trainer = mlContext.Regression.Trainers.LightGbm(options);
var trainingPipeline = dataProcessPipeline.Append(trainer);
var trainedModel = trainingPipeline.Fit(folds[i].TrainSet);    //No Fit(train, validationDv) Method 
```

**Comment by rauhs** [comment]
Correct, see #3603
